xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1129 @@
+        if self.arbiterID.type != IntegerType(8):
+            raise VerifyException("arbiterID has to be an 8-bit signless integer")
+        if self.arbiterID.value.data < 0 or self.arbiterID.value.data > 5:
+            raise VerifyException("arbiterID has to be in the range [0-5].")
+        if self.msel.type != IntegerType(8):
+            raise VerifyException("msel has to be an 8-bit signless integer")
+        if self.msel.value.data < 0 or self.arbiterID.value.data > 3:
+            raise VerifyException("msel has to be in the range [0-3].")
+
+
+@irdl_op_definition
+class BufferOp(IRDLOperation):
+    name = "AIE.buffer"
+    tile = operand_def(IndexType())
+    shape = attr_def(ArrayAttr[IntegerType])
+    element_type = attr_def(Attribute)
+    buffer = result_def(memref.MemRefType)
+    sym_name = attr_def(StringAttr)
+
+    def __init__(
+        self,
+        tile: Operation | SSAValue,
+        element_type: Attribute,
+        shape: ArrayAttr[AnyIntegerAttr],
+        sym_name: StringAttr,
+    ):
+        buffer_type = memref.MemRefType.from_element_type_and_shape(element_type, shape)
+        super().__init__(
+            operands=[tile],
+            attributes={"sym_name": sym_name},
+            result_types=[buffer_type],
+        )
+
+
+@irdl_op_definition
+class TileOp(IRDLOperation):
+    name = "AIE.tile"
+    col = attr_def(IntegerAttr[IntegerType])
+    row = attr_def(IntegerAttr[IntegerType])
+    result = result_def(IndexType())
+
+    def __init__(self, col: IntegerAttr[IntegerType], row: IntegerAttr[IntegerType]):
+        super().__init__(
+            attributes={"col": col, "row": row}, result_types=[IndexType()]
+        )
+
+
+@irdl_op_definition
+class ConnectOp(IRDLOperation):
+    name = "AIE.connect"
+    sourceBundle = attr_def(WireBundleAttr)
+    sourceChannel = attr_def(AnyIntegerAttr)
+    destBundle = attr_def(WireBundleAttr)
+    destChannel = attr_def(AnyIntegerAttr)
+
+    def __init__(
+        self,
+        sourceBundle: WireBundleAttr,
+        sourceChannel: int | AnyIntegerAttr,
+        destBundle: WireBundleAttr,
+        destChannel: int | AnyIntegerAttr,
+    ):
+        if isinstance(sourceChannel, int):
+            sourceChannel = IntegerAttr(sourceChannel, IntegerType(8))
+        if isinstance(destChannel, int):
+            destChannel = IntegerAttr(destChannel, IntegerType(8))
+        super().__init__(
+            attributes={
+                "sourceBundle": sourceBundle,
+                "sourceChannel": sourceChannel,
+                "destBundle": destBundle,
+                "destChannel": destChannel,
+            }
+        )
+
+    def print(self, printer: Printer):
+        printer.print(
+            "<",
+            self.sourceBundle.data,
+            ": ",
+            self.sourceChannel.value.data,
+            ", ",
+            self.destBundle.data,
+            ": ",
+            self.destChannel.value.data,
+            ">",
+        )
```

Comment:
This is missing a corresponding `parse` function. But also, it feels like this could leverage the format description API instead, like so:
```suggestion
    def print(self, printer: Printer):
        printer.print(
            "<",
            self.sourceBundle.data,
            ": ",
            self.sourceChannel.value.data,
            ", ",
            self.destBundle.data,
            ": ",
            self.destChannel.value.data,
            ">",
        )
        assembly_format = "`<` $sourceBundle `: ` $sourceChannel `, ` $destBundle `: ` destChannel `>`  attr-dict"
```

Reply:
I'm getting an error when using the format description API. It's my first time using it, but from what I've seen in other dialects just by removing the print method and having the `assembly_format` definition you provide it should be fine. I get an error during parser though: `'expected variable to refer to an operand, attribute, region, result, or successor'`. It fails with the first variable already, `sourceBundle`.

Reply:
Can I have a go at this and push to your PR?

Reply:
Ah I spoke with @math-fehr, and it turns out that there isn't support for attributes right now. He says that maybe he can work on it today or tomorrow, not sure if it's worth blocking the PR on this, even if it would make implementing the printers and parsers easier.

Reply:
If it just has to wait one day maybe I'd say it's worth the wait. I have to work on some details of the execution of the code on actual hardware anyways, as it needs some tweaks with respect to the simulator.

Reply:
Let's see what @math-fehr says

Reply:
#2028

Reply:
Just merged main with this branch and I keep getting the same error.

Reply:
Can you please push the broken version?

Reply:
@superlopuh sorry, didn't get a notification of this! The broken version is already pushed.

Reply:
It looks like all the tests pass, there's no error. Could you please push the filecheck also?

Reply:
Sorry, I didn't commit the broken version in a "broken" state, but with the custom print method and I thought you would replace it with the API line on your end as I was doing! Just comitted the broken version and now you can see how a test fails and gives the same error I get.

Reply:
Thank you, I merged main and fixed a few issues. I also added a super basic filecheck test. It feels like it would be worth the effort to add the other ops for which you'd like to have custom syntax to the filecheck, just to make sure that they roundtrip ok